python/ray/util/collective/collective.py

```diff
@@ -609,21 +627,54 @@ def send_multigpu(
     Returns:
         None
     """
-    if not types.cupy_available():
-        raise RuntimeError("send_multigpu call requires NCCL.")
+    send_multidevice(
+        tensor=tensor,
+        dst_rank=dst_rank,
+        dst_device_index=dst_gpu_index,
+        group_name=group_name,
+        n_elements=n_elements,
+    )
+
+
+def send_multidevice(
```

Comment:
Can you leave these changes to a separate PR? It will be easier to merge if we can split this up into smaller PRs, and API changes also require closer review.